### ISSUE TITLE
Crash ``'Attribute' object has no attribute 'name'`` in _check_unused_private_variables due to access via self.__class__

### ISSUE THREAD

**Issue by graingert** [PyCQA/pylint#5261]
When parsing the following file:

```python
class Foo:
    __ham = 1

    def foo():
        print(self.__class__.__ham)

```

pylint crashed with a ``AttributeError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "/home/graingert/.local/pipx/.cache/1ea90594a10eda9/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1008, in _check_files
    self._check_file(get_ast, check_astroid_module, file)
  File "/home/graingert/.local/pipx/.cache/1ea90594a10eda9/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1043, in _check_file
    check_astroid_module(ast_node)
  File "/home/graingert/.local/pipx/.cache/1ea90594a10eda9/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1180, in check_astroid_module
    retval = self._check_astroid_module(
  File "/home/graingert/.local/pipx/.cache/1ea90594a10eda9/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1227, in _check_astroid_module
    walker.walk(node)
  File "/home/graingert/.local/pipx/.cache/1ea90594a10eda9/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 78, in walk
    self.walk(child)
  File "/home/graingert/.local/pipx/.cache/1ea90594a10eda9/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 80, in walk
    callback(astroid)
  File "/home/graingert/.local/pipx/.cache/1ea90594a10eda9/lib/python3.8/site-packages/pylint/checkers/classes.py", line 907, in leave_classdef
    self._check_unused_private_variables(node)
  File "/home/graingert/.local/pipx/.cache/1ea90594a10eda9/lib/python3.8/site-packages/pylint/checkers/classes.py", line 973, in _check_unused_private_variables
    and child.expr.name in ("self", "cls", node.name)
AttributeError: 'Attribute' object has no attribute 'name'
```

**Comment by graingert** [comment]
this code found in the wild at https://github.com/django-treebeard/django-treebeard/blob/8eb5082e67c05057710e8e07365059c99dce47e5/treebeard/forms.py#L91-L94

**Comment by DanielNoord** [comment]
Thanks for the report!

We should be able to ship a fix with `2.12` which I think will be published soon™.

**Comment by graingert** [comment]
> Thanks for the report!
> 
> We should be able to ship a fix with `2.12` which I think will be published soon™.

Thanks for this, Amazing speed in getting this fixed!
